### PR TITLE
fix no match

### PR DESCRIPTION
**Pull request by yuval-k**

### CODE REVIEW

File: projects/gateway2/translator/testutils/inputs/https-routing/gateway.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+#$ Used in:
+#$ - site-src/guides/http-routing.md
```
Comment (lgadban) [inline projects/gateway2/translator/testutils/inputs/https-routing/gateway.yaml:2]: is this the actual example from the gwAPI docs?

File: projects/gateway2/translator/httproute/gateway_http_route_translator.go [hunk 2]
```diff
@@ -33,6 +33,14 @@ func TranslateGatewayHTTPRouteRules(
 ) []*routeutils.SortableRoute {
 	var outputSortableRoutes []*routeutils.SortableRoute
 	for i, rule := range route.Spec.Rules {
+
+		rule := rule
+		if rule.Matches == nil {
+			// from the spec:
+			// If no matches are specified, the default is a prefix path match on “/”, which has the effect of matching every HTTP request.
```
Comment (EItanya) [inline projects/gateway2/translator/httproute/gateway_http_route_translator.go:40]: Breaks `DeepCopy` 😆

File: projects/gateway2/translator/testutils/inputs/https-routing/gateway.yaml (hunk 1 above)
Comment (yuval-k) [reply projects/gateway2/translator/testutils/inputs/https-routing/gateway.yaml:2]: i forget where i got this from :/ why?

File: projects/gateway2/translator/httproute/gateway_http_route_translator.go (hunk 2 above)
Comment (yuval-k) [reply projects/gateway2/translator/httproute/gateway_http_route_translator.go:40]: it doesn't as the rule is copied just before the `if`

**Review by EItanya — APPROVED** [review]

File: projects/gateway2/translator/testutils/inputs/https-routing/gateway.yaml (hunk 1 above)
Comment (lgadban) [reply projects/gateway2/translator/testutils/inputs/https-routing/gateway.yaml:2]: don't remember why I asked either; not important !

**Pull request merged**